creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -555,83 +406,35 @@ class SnapbackSM {
   async processStateMachineOperation () {
     // Record all stages of this function along with associated information for use in logging
     let decisionTree = [{
-      stage: '1/ BEGIN processStateMachineOperation()',
+      stage: 'BEGIN processStateMachineOperation()',
       vals: {
         currentModuloSlice: this.currentModuloSlice
       },
       time: Date.now()
     }]
 
+    let nodeUsers
     try {
-      /**
-       * Retrieve list of all users which have this node as replica (primary or secondary) from discovery node
-       * Or retrieve primary users only if connected to old discprov
-       */
-      let nodeUsers
-      try {
-        nodeUsers = await this.getNodeUsers(decisionTree)
-        decisionTree.push({ stage: '2/ getNodeUsers() Success', vals: { nodeUsersLength: nodeUsers.length }, time: Date.now() })
-      } catch (e) {
-        decisionTree.push({ stage: '2/ getNodeUsers() Error', vals: e.message, time: Date.now() })
-        throw new Error('processStateMachineOperation():getAllNodeUsers() Error')
-      }
-
-      /**
-       * Select chunk of users to process in this run
-       *  - User is selected if (user_id % ModuloBase = currentModuloSlice)
-       *  - nodeUsers = array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
-       */
-      nodeUsers = nodeUsers.filter(
-        nodeUser => (nodeUser.user_id % ModuloBase === this.currentModuloSlice)
-      )
-      decisionTree.push({
-        stage: '3/ select nodeUsers modulo slice',
-        vals: { nodeUsersSubsetLength: nodeUsers.length, moduloBase: ModuloBase, currentModuloSlice: this.currentModuloSlice },
-        time: Date.now()
-      })
-
-      // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
-      let peerSet = []
-      try {
-        peerSet = this.computeContentNodePeerSet(nodeUsers)
+      nodeUsers = await this.peerSetManager.getNodeUsers()
```

Comment:
also any reason to move the initial steps (lines 416-435) out of the global try-catch? that try-catch-finally is important bc it ensures the decisionTree state is logged out always (see lines 596-601)

Reply:
no, this was just an oversight in my PR. moving